generate_figures/utilities.py

```diff
@@ -0,0 +1,55 @@
+import pandas as pd
+import wandb
+
+def plot_params(figsize, fontsize, ticksize):
+    params = {
+        "text.usetex": True,
+        "font.family": "serif",
+        "figure.figsize": figsize,
+        "figure.dpi": 80,
+        "figure.edgecolor": "k",
+        "font.size": fontsize,
+        "axes.labelsize": fontsize,
+        "axes.titlesize": fontsize,
+        "xtick.labelsize": ticksize,
+        "ytick.labelsize": ticksize,
+    }
+    return(params)
+
+
+
+def get_results_by_tag(project, tag, test_results = False):
+    api = wandb.Api()
+    count=0
+    runs = api.runs(project, filters={"tags": tag})
+    df_test_results = pd.DataFrame()
+    for i in range(len(runs)):
+        count+=1
+        df = pd.DataFrame()
+        cols = dict(runs[i].summary)
+        cols.update(dict(runs[i].config))
+        cols['id'] = runs[i].id
+        df = pd.DataFrame.from_dict(cols, orient='index').T
+ 
+        if test_results:
+            reference_path = f"{project}/run-{cols['id']}-test_results:v0"
+            test_results = api.artifact(str(reference_path)).get("test_results")
+            data = pd.DataFrame(data = test_results.data, columns = test_results.columns)
+            df = pd.concat([df, data], axis = 1)
+            df.fillna(method='ffill', inplace=True)
```

Comment:
This scares me a little.  Why do we want to change any of the NaN?  Seems like we want to know which ones failed and which didnt?

Reply:
@jlperla ohhh okay so its because the summary stuff is for all the runs so we have like 64 or something in the rows for the artifact but only one row for the summary/config data so I wanted to fill the table because it was just predominately nan's but you are very right because I need to only do this for the columns that were part of the summary/config and not replace the nan's in the artifact part

Reply:
@jlperla or maybe its fine to have the table filled with a ton of nans under the config details so something like 'seed' would only have row 0 have a value and everything else is nan which I guess would be fine

Reply:
Ah this was the thing Iwas pointing out about it being different from the previous cod.e  Let me fix it and you can play around with it after.